addons/sale_project/models/sale_order_line.py

```diff
@@ -43,7 +43,7 @@ def default_get(self, fields):
                     try:
                         project_so = self.env['project.project'].browse(project_id).sale_order_id
                         project_so.check_access('write')
-                        sale_order = project_so
+                        sale_order = project_so or self.env['sale.order'].search([('project_id', '=', project_id)], limit=1, order='date_order DESC')
```

Comment:
The order is not needed since the order defined on `sale.order` is `date_order desc, id desc`, see https://github.com/odoo/odoo/blob/b980ad0e2fec8258b7674ac2490ac2abe716841b/addons/sale/models/sale_order.py#L48

```suggestion
                        sale_order = project_so or self.env['sale.order'].search([('project_id', '=', project_id)], limit=1)
```
🙂